flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/spargel/VertexCentricIteration.java

```diff
@@ -228,48 +213,92 @@ public void setInput(DataSet<Vertex<VertexKey, VertexValue>> inputData) {
 	{
 		return new VertexCentricIteration<VertexKey, VertexValue, Message, EdgeValue>(uf, mf, edgesWithValue, maximumNumberOfIterations);
 	}
-	
+
+	/**
+	 * Configures this vertex-centric iteration with the provided parameters.
+	 *
+	 * @param parameters the configuration parameters
+	 */
+	public void configure(IterationConfiguration parameters) {
+		this.configuration = parameters;
+	}
+
+	/**
+	 * @return the configuration parameters of this vertex-centric iteration
+	 */
+	public IterationConfiguration getIterationConfiguration() {
+		return this.configuration;
+	}
+
 	// --------------------------------------------------------------------------------------------
 	//  Wrapping UDFs
 	// --------------------------------------------------------------------------------------------
 	
-	private static final class VertexUpdateUdf<VertexKey extends Comparable<VertexKey> & Serializable, 
-		VertexValue extends Serializable, Message> 
-		extends RichCoGroupFunction<Tuple2<VertexKey, Message>, Vertex<VertexKey, VertexValue>, Vertex<VertexKey, VertexValue>>
-		implements ResultTypeQueryable<Vertex<VertexKey, VertexValue>>
+	private static abstract class VertexUpdateUdf<VertexKey extends Comparable<VertexKey> & Serializable,
+		VV extends Serializable, VertexValue extends Serializable, Message>
```

Comment:
Perhaps I chose a name that is a bit misleading, here VV is replaced by the Tuple3<VertexValue, InDegree, OutDegree> when you have the  degrees option set or simply VertexValue if the degrees are not set. I still need the VertexValue generic type. I can rename VertexValue to VV, EdgeValue to EV, etc and VV to something else? Something like  VertexValueMessagingFunction. I can't think of a better name :) 

Tell me if my intention is clearer now and how to proceed!